### PR TITLE
Connected the assignUser function for user-selector

### PR DESCRIPTION
**Pull request by Mr-Smyth**
Connected the assignUser function for user-selector to task controller also removed task user ref in json:

- added a tasks controller to handle the task / user assignment - used set to create the new association. Noticed Save() didn't work this seems to trigger a save action to an API or whatever data source is setup and that caused errors as im using a json file for demo data.
- Removed the user ref on the tasks json file: [public/api/tasks.json](url)
- added file path names to tops of files for my own clarity
- added/edited comments
- changed style of separated unassigned tasks 
- Reformatted files

### CODE REVIEW

**Comment by asilbekcodes** [comment]
hello

File: app/routes/tasks.js [hunk 1]
```diff
@@ -1,67 +1,61 @@
...
+      if (user) {
+        let userName = user.get('name');
+        if (userName !== 'Unassigned') {
+          // If name is  not 'Unassigned', return true to include this task in Assigned
+          return true;
+        }
+      }
+      // If any condition fails, return false to skip this task
+      return false;
+    });
```
Comment (far-fetched) [inline app/routes/tasks.js]: As discussed, these 2 methods are perfect candidates to getter into controller / component.

File: app/controllers/tasks.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    // Retrieve the task and user records by their IDs
+    let task = this.store.peekRecord('task', taskId);
+    let user = this.store.peekRecord('user', userId);
+
+    // If both task and user exist, assign the user to the task use set
+    // 
+    if (task && user) {
+      task.set('user', user);
+      this.router.refresh('tasks');
```
Comment (far-fetched) [inline app/controllers/tasks.js]: Ideally, we would like to remove this “refresh” method at all.

**Comment by Mr-Smyth** [comment]
I couldnt get it to work correctly after removing the refresh('tasks'). I think this could have been to do with the way I was passing the already unclean data from my model. So I eventually decided to skip onto moving to a clean model export from the route and then worked on redoing the controller. This approach meant that I just didn't need the refresh, and once the controller was setup to work correctly it now refreshes the page content automatically when I reassign a task to a different user. Also changed my CSS a little as the bobbing of the tasks on hover was really bugging me

File: app/controllers/tasks.js (hunk 2 above)
Comment (Mr-Smyth) [reply app/controllers/tasks.js]: Done

File: app/routes/tasks.js (hunk 1 above)
Comment (Mr-Smyth) [reply app/routes/tasks.js]: Done - moved to the tasks controller

**Comment by aslbekDev07** [comment]
hello

**Review by far-fetched — APPROVED** [review]

**Pull request merged**